### ISSUE TITLE
[Next-plugin] Cannot read properties of undefined (reading 'watcher')

### ISSUE THREAD

**Issue by conarwelsh** [facebook/stylex#288]
**The problem**

I am constantly getting the same error:
```
./src/styles/tokens.stylex.js
TypeError: Cannot read properties of undefined (reading 'watcher')
    at Set.forEach (<anonymous>)
    at Set.forEach (<anonymous>)
```
It seems to happen randomly after saving any file, does not matter what file.  I was waiting to file a ticket on it until i could get a more predictable way to reproduce the error, but I can't really pinpoint it.

However, if I just save the file again, the auto refresh often will clear the error and it will be fine.  Or if that doesn't work I can restart the dev server and that often fixes it.  Sometimes it is more persistant than others.  Sometimes I have to save the file and other files many times, and restart the server several times before the error clears (without changing a single line of code).

Also I notice that if I hit any JS errors at all completely unrelated to stylex, for instance a typo or bad syntax or whatever, instead of showing me the JS errors i get shown
```
TypeError: Failed to set an indexed property [0] on 'CSSStyleDeclaration': Indexed property setter is not supported.
```

**How to reproduce**

1. Using the nextjs plugin, save a file
2. the error will appear
3. save again
4. error sometimes disappears

I am not sure what info or setup information or whatever you need to debug but am happy to provide whatever.  For starters:

```
"@stylexjs/open-props": "^0.4.1",
"@stylexjs/stylex": "^0.4.1",
"@stylexjs/babel-plugin": "^0.4.1",
"@stylexjs/eslint-plugin": "^0.4.1",
"@stylexjs/nextjs-plugin": "^0.4.1",
```

```js
const stylexPlugin = require("@stylexjs/nextjs-plugin")

const nextConfig = {
  pageExtensions: ["js", "jsx", "ts", "tsx"],
  reactStrictMode: true,
  swcMinify: true,
  compiler: {
    removeConsole: process.env.NODE_ENV !== "development",
  },
}

module.exports = stylexPlugin({
  rootDir: __dirname,
})(nextConfig)
```

```js
module.exports = {
  presets: ["next/babel"],
  plugins: [
    [
      "@stylexjs/babel-plugin",
      {
        dev: process.env.NODE_ENV === "development",
        runtimeInjection: false,
        genConditionalClasses: true,
        treeshakeCompensation: true,
        unstable_moduleResolution: {
          type: "commonJS",
          rootDir: __dirname,
        },
      },
    ],
  ],
}
```

```js
import * as stylex from "@stylexjs/stylex"

const DARK = "@media (prefers-color-scheme: dark)"
const MOBILE = "@media (max-width: 700px)"

const MIN_WIDTH = 320
const MAX_WIDTH = 1240
const MIN_SCALE = 1.2
const MAX_SCALE = 1.333
const MIN_BASE_SIZE = 10 //16
const MAX_BASE_SIZE = 18 //20

// intentionally wrong colors to spot errors with themes
export const colors = stylex.defineVars({
  primaryText: "lime",
  primaryTextInverted: "lime",
  secondaryText: "lime",
  secondaryTextInverted: "lime",
  lightText: "lime",
  darkText: "lime",
  surface1: "lime",
  surface2: "lime",
  surface3: "lime",
  surface4: "lime",
  border: "lime",
  primary: "lime",
  primary1: "lime",
  primary2: "lime",
  primary3: "lime",
  primary4: "lime",
  secondary: "lime",
  secondary1: "lime",
  secondary2: "lime",
  secondary3: "lime",
  secondary4: "lime",
  info: "lime",
  warning: "lime",
  success: "lime",
  error: "lime",
})

export const shadowVars = stylex.defineVars({
  shadowColor: {
    default: "220 3% 15%",
    [DARK]: "220 40% 2%",
  },
  shadowStrength: {
    default: "1%",
    [DARK]: "25%",
  },
  innerShadowHighlight: {
    default: "inset 0 -.5px 0 0 #fff, inset 0 .5px 0 0 #0001",
    [DARK]: "inset 0 -.5px 0 0 #fff1, inset 0 .5px 0 0 #0007",
  },
})

export const shadows = stylex.defineVars({
  shadow1: `0 1px 2px -1px hsl(${shadowVars.shadowColor} / calc(${shadowVars.shadowStrength} + 9%))`,
  shadow2: `0 3px 5px -2px hsl(${shadowVars.shadowColor} / calc(${shadowVars.shadowStrength} + 3%)),
    0 7px 14px -5px hsl(${shadowVars.shadowColor} / calc(${shadowVars.shadowStrength} + 5%))`,
  shadow3: `0 -1px 3px 0 hsl(${shadowVars.shadowColor} / calc(${shadowVars.shadowStrength} + 2%)),
    0 1px 2px -5px hsl(${shadowVars.shadowColor} / calc(${shadowVars.shadowStrength} + 2%)),
    0 2px 5px -5px hsl(${shadowVars.shadowColor} / calc(${shadowVars.shadowStrength} + 4%)),
    0 4px 12px -5px hsl(${shadowVars.shadowColor} / calc(${shadowVars.shadowStrength} + 5%)),
    0 12px 15px -5px hsl(${shadowVars.shadowColor} / calc(${shadowVars.shadowStrength} + 7%))`,
  shadow4: `0 -2px 5px 0 hsl(${shadowVars.shadowColor} / calc(${shadowVars.shadowStrength} + 2%)),
    0 1px 1px -2px hsl(${shadowVars.shadowColor} / calc(${shadowVars.shadowStrength} + 3%)),
    0 2px 2px -2px hsl(${shadowVars.shadowColor} / calc(${shadowVars.shadowStrength} + 3%)),
    0 5px 5px -2px hsl(${shadowVars.shadowColor} / calc(${shadowVars.shadowStrength} + 4%)),
    0 9px 9px -2px hsl(${shadowVars.shadowColor} / calc(${shadowVars.shadowStrength} + 5%)),
    0 16px 16px -2px hsl(${shadowVars.shadowColor} / calc(${shadowVars.shadowStrength} + 6%))`,
  shadow5: `0 -1px 2px 0 hsl(${shadowVars.shadowColor} / calc(${shadowVars.shadowStrength} + 2%)),
    0 2px 1px -2px hsl(${shadowVars.shadowColor} / calc(${shadowVars.shadowStrength} + 3%)),
    0 5px 5px -2px hsl(${shadowVars.shadowColor} / calc(${shadowVars.shadowStrength} + 3%)),
    0 10px 10px -2px hsl(${shadowVars.shadowColor} / calc(${shadowVars.shadowStrength} + 4%)),
    0 20px 20px -2px hsl(${shadowVars.shadowColor} / calc(${shadowVars.shadowStrength} + 5%)),
    0 40px 40px -2px hsl(${shadowVars.shadowColor} / calc(${shadowVars.shadowStrength} + 7%))`,
  shadow6: `0 -1px 2px 0 hsl(${shadowVars.shadowColor} / calc(${shadowVars.shadowStrength} + 2%)),
    0 3px 2px -2px hsl(${shadowVars.shadowColor} / calc(${shadowVars.shadowStrength} + 3%)),
    0 7px 5px -2px hsl(${shadowVars.shadowColor} / calc(${shadowVars.shadowStrength} + 3%)),
    0 12px 10px -2px hsl(${shadowVars.shadowColor} / calc(${shadowVars.shadowStrength} + 4%)),
    0 22px 18px -2px hsl(${shadowVars.shadowColor} / calc(${shadowVars.shadowStrength} + 5%)),
    0 41px 33px -2px hsl(${shadowVars.shadowColor} / calc(${shadowVars.shadowStrength} + 6%)),
    0 100px 80px -2px hsl(${shadowVars.shadowColor} / calc(${shadowVars.shadowStrength} + 7%))`,
  innerShadow0: `inset 0 0 0 1px hsl(${shadowVars.shadowColor} / calc(${shadowVars.shadowStrength} + 9%))`,
  innerShadow1: `inset 0 1px 2px 0 hsl(${shadowVars.shadowColor} / calc(${shadowVars.shadowStrength} + 9%)), ${shadowVars.innerShadowHighlight}`,
  innerShadow2: `inset 0 1px 4px 0 hsl(${shadowVars.shadowColor} / calc(${shadowVars.shadowStrength} + 9%)), ${shadowVars.innerShadowHighlight}`,
  innerShadow3: `inset 0 2px 8px 0 hsl(${shadowVars.shadowColor} / calc(${shadowVars.shadowStrength} + 9%)), ${shadowVars.innerShadowHighlight}`,
  innerShadow4: `inset 0 2px 14px 0 hsl(${shadowVars.shadowColor} / calc(${shadowVars.shadowStrength} + 9%)), ${shadowVars.innerShadowHighlight}`,
})

/**
 * o--o o    o   o o-O-o o-o       o--o  o-o  o   o o-O-o  o-o
 * |    |    |   |   |   |  \      |    o   o |\  |   |   |
 * O-o  |    |   |   |   |   O     O-o  |   | | \ |   |    o-o
 * |    |    |   |   |   |  /      |    o   o |  \|   |       |
 * o    O---o o-o  o-O-o o-o       o     o-o  o   o   o   o--o
 *
 * Reference: https://utopia.fyi/type/calculator
 *
 * The following constants are used to calculate fluid typography.
 * Feel free to change these initial numbers to suit your needs.
 *
 * StyleX can compute all of this at compile time as all the information
 * is statically available in the same file and the only functions used are
 * the Math.pow and Math.round functions.
 *
 * NOTE: Any custom functions will not be able to be computed at compile time.
 */

// Font sizes in `rem` units
const MIN_FONT = {
  xxs: Math.round(MIN_BASE_SIZE / Math.pow(MIN_SCALE, 3) / 0.16) / 100,
  xs: Math.round(MIN_BASE_SIZE / Math.pow(MIN_SCALE, 2) / 0.16) / 100,
  sm: Math.round(MIN_BASE_SIZE / MIN_SCALE / 0.16) / 100,
  p: Math.round(MIN_BASE_SIZE / 4) / 4,
  h5: Math.round((MIN_BASE_SIZE * MIN_SCALE) / 0.16) / 100,
  h4: Math.round((MIN_BASE_SIZE * Math.pow(MIN_SCALE, 2)) / 0.16) / 100,
  h3: Math.round((MIN_BASE_SIZE * Math.pow(MIN_SCALE, 3)) / 0.16) / 100,
  h2: Math.round((MIN_BASE_SIZE * Math.pow(MIN_SCALE, 4)) / 0.16) / 100,
  h1: Math.round((MIN_BASE_SIZE * Math.pow(MIN_SCALE, 5)) / 0.16) / 100,
}
// Font sizes in `rem` units
const MAX_FONT = {
  xxs: Math.round(MAX_BASE_SIZE / Math.pow(MAX_SCALE, 3) / 0.16) / 100,
  xs: Math.round(MAX_BASE_SIZE / Math.pow(MAX_SCALE, 2) / 0.16) / 100,
  sm: Math.round(MAX_BASE_SIZE / MAX_SCALE / 0.16) / 100,
  p: Math.round(MAX_BASE_SIZE / 4) / 4,
  h5: Math.round((MAX_BASE_SIZE * MAX_SCALE) / 0.16) / 100,
  h4: Math.round((MAX_BASE_SIZE * Math.pow(MAX_SCALE, 2)) / 0.16) / 100,
  h3: Math.round((MAX_BASE_SIZE * Math.pow(MAX_SCALE, 3)) / 0.16) / 100,
  h2: Math.round((MAX_BASE_SIZE * Math.pow(MAX_SCALE, 4)) / 0.16) / 100,
  h1: Math.round((MAX_BASE_SIZE * Math.pow(MAX_SCALE, 5)) / 0.16) / 100,
}
const SLOPE = {
  xxs: (16 * (MAX_FONT.xxs - MIN_FONT.xxs)) / (MAX_WIDTH - MIN_WIDTH),
  xs: (16 * (MAX_FONT.xs - MIN_FONT.xs)) / (MAX_WIDTH - MIN_WIDTH),
  sm: (16 * (MAX_FONT.sm - MIN_FONT.sm)) / (MAX_WIDTH - MIN_WIDTH),
  p: (16 * (MAX_FONT.p - MIN_FONT.p)) / (MAX_WIDTH - MIN_WIDTH),
  h5: (16 * (MAX_FONT.h5 - MIN_FONT.h5)) / (MAX_WIDTH - MIN_WIDTH),
  h4: (16 * (MAX_FONT.h4 - MIN_FONT.h4)) / (MAX_WIDTH - MIN_WIDTH),
  h3: (16 * (MAX_FONT.h3 - MIN_FONT.h3)) / (MAX_WIDTH - MIN_WIDTH),
  h2: (16 * (MAX_FONT.h2 - MIN_FONT.h2)) / (MAX_WIDTH - MIN_WIDTH),
  h1: (16 * (MAX_FONT.h1 - MIN_FONT.h1)) / (MAX_WIDTH - MIN_WIDTH),
}
const INTERCEPT = {
  xxs: Math.round(100 * (MIN_FONT.xxs - SLOPE.xxs * (MIN_WIDTH / 16))) / 100,
  xs: Math.round(100 * (MIN_FONT.xs - SLOPE.xs * (MIN_WIDTH / 16))) / 100,
  sm: Math.round(100 * (MIN_FONT.sm - SLOPE.sm * (MIN_WIDTH / 16))) / 100,
  p: Math.round(100 * (MIN_FONT.p - SLOPE.p * (MIN_WIDTH / 16))) / 100,
  h5: Math.round(100 * (MIN_FONT.h5 - SLOPE.h5 * (MIN_WIDTH / 16))) / 100,
  h4: Math.round(100 * (MIN_FONT.h4 - SLOPE.h4 * (MIN_WIDTH / 16))) / 100,
  h3: Math.round(100 * (MIN_FONT.h3 - SLOPE.h3 * (MIN_WIDTH / 16))) / 100,
  h2: Math.round(100 * (MIN_FONT.h2 - SLOPE.h2 * (MIN_WIDTH / 16))) / 100,
  h1: Math.round(100 * (MIN_FONT.h1 - SLOPE.h1 * (MIN_WIDTH / 16))) / 100,
}

export const text = stylex.defineVars({
  sans: "'Montserrat', sans-serif",
  script: "'Marck Script',cursive",
  thin: 100,
  light: 300,
  bold: 600,
  xxs: `clamp(${Math.min(MIN_FONT.xxs)}rem, calc(${INTERCEPT.xxs}rem + ${
    Math.round(10000 * SLOPE.xxs) / 100
  }vw), ${Math.max(MAX_FONT.xxs)}rem)`,
  xs: `clamp(${Math.min(MIN_FONT.xs)}rem, calc(${INTERCEPT.xs}rem + ${
    Math.round(10000 * SLOPE.xs) / 100
  }vw), ${Math.max(MAX_FONT.xs)}rem)`,
  sm: `clamp(${Math.min(MIN_FONT.sm)}rem, calc(${INTERCEPT.sm}rem + ${
    Math.round(10000 * SLOPE.sm) / 100
  }vw), ${Math.max(MAX_FONT.sm)}rem)`,
  p: `clamp(${Math.min(MIN_FONT.p)}rem, calc(${INTERCEPT.p}rem + ${
    Math.round(10000 * SLOPE.p) / 100
  }vw), ${Math.max(MAX_FONT.p)}rem)`,
  h5: `clamp(${Math.min(MIN_FONT.h5)}rem, calc(${INTERCEPT.h5}rem + ${
    Math.round(10000 * SLOPE.h5) / 100
  }vw), ${Math.max(MAX_FONT.h5)}rem)`,
  h4: `clamp(${Math.min(MIN_FONT.h4)}rem, calc(${INTERCEPT.h4}rem + ${
    Math.round(10000 * SLOPE.h4) / 100
  }vw), ${Math.max(MAX_FONT.h4)}rem)`,
  h3: `clamp(${Math.min(MIN_FONT.h3)}rem, calc(${INTERCEPT.h3}rem + ${
    Math.round(10000 * SLOPE.h3) / 100
  }vw), ${Math.max(MAX_FONT.h3)}rem)`,
  h2: `clamp(${Math.min(MIN_FONT.h2)}rem, calc(${INTERCEPT.h2}rem + ${
    Math.round(10000 * SLOPE.h2) / 100
  }vw), ${Math.max(MAX_FONT.h2)}rem)`,
  h1: `clamp(${Math.min(MIN_FONT.h1)}rem, calc(${INTERCEPT.h1}rem + ${
    Math.round(10000 * SLOPE.h1) / 100
  }vw), ${Math.max(MAX_FONT.h1)}rem)`,
})

/**
 * o--o o    o   o o-O-o o-o        o-o  o--o    O    o-o o--o
 * |    |    |   |   |   |  \      |     |   |  / \  /    |
 * O-o  |    |   |   |   |   O      o-o  O--o  o---oO     O-o
 * |    |    |   |   |   |  /          | |     |   | \    |
 * o    O---o o-o  o-O-o o-o       o--o  o     o   o  o-o o--o
 *
 * Reference: https://utopia.fyi/space/calculator
 *
 * Similar to the fluid typography, we can create fluid values for spacing.
 * Using similar formulas and similar scales.
 *
 * NOTE: It is common to have more varied needs for spacing than for font-size.
 * So feel free to add some more values by following the pattern below.
 *
 * EXCEPT: We are using `px` instead of `rem`
 * ------------------------------------------
 * When talking about font-size, it is the best practice to use
 * `rem` so that an end user can change the font-size using the
 * browser's font-size setting.
 *
 * However, when talking about spacing, it is the best practice to
 * use `px` because using `rems` here makes font-size behave like zoom.
 *
 * Users that prefer larger text, don't neccessarily want larger spacing as well.
 *
 */

const MULT = {
  xxxs: 0.25,
  xxs: 0.5,
  xs: 0.75,
  sm: 1,
  md: 1.5,
  lg: 2,
  xl: 3,
  xxl: 4,
  xxxl: 6,
  xxxxl: 8,
}
const MIN_SPACE = {
  xxxs: MULT.xxxs * MIN_BASE_SIZE,
  xxs: MULT.xxs * MIN_BASE_SIZE,
  xs: MULT.xs * MIN_BASE_SIZE,
  sm: MULT.sm * MIN_BASE_SIZE,
  md: MULT.md * MIN_BASE_SIZE,
  lg: MULT.lg * MIN_BASE_SIZE,
  xl: MULT.xl * MIN_BASE_SIZE,
  xxl: MULT.xxl * MIN_BASE_SIZE,
  xxxl: MULT.xxxl * MIN_BASE_SIZE,
  xxxxl: MULT.xxxxl * MIN_BASE_SIZE,
}
const MAX_SPACE = {
  xxxs: MULT.xxxs * MAX_BASE_SIZE,
  xxs: MULT.xxs * MAX_BASE_SIZE,
  xs: MULT.xs * MAX_BASE_SIZE,
  sm: MULT.sm * MAX_BASE_SIZE,
  md: MULT.md * MAX_BASE_SIZE,
  lg: MULT.lg * MAX_BASE_SIZE,
  xl: MULT.xl * MAX_BASE_SIZE,
  xxl: MULT.xxl * MAX_BASE_SIZE,
  xxxl: MULT.xxxl * MAX_BASE_SIZE,
  xxxxl: MULT.xxxxl * MAX_BASE_SIZE,
}
const SLOPE_SPACE = {
  xxxs: (MAX_SPACE.xxxs - MIN_SPACE.xxxs) / (MAX_WIDTH - MIN_WIDTH),
  xxs: (MAX_SPACE.xxs - MIN_SPACE.xxs) / (MAX_WIDTH - MIN_WIDTH),
  xs: (MAX_SPACE.xs - MIN_SPACE.xs) / (MAX_WIDTH - MIN_WIDTH),
  sm: (MAX_SPACE.sm - MIN_SPACE.sm) / (MAX_WIDTH - MIN_WIDTH),
  md: (MAX_SPACE.md - MIN_SPACE.md) / (MAX_WIDTH - MIN_WIDTH),
  lg: (MAX_SPACE.lg - MIN_SPACE.lg) / (MAX_WIDTH - MIN_WIDTH),
  xl: (MAX_SPACE.xl - MIN_SPACE.xl) / (MAX_WIDTH - MIN_WIDTH),
  xxl: (MAX_SPACE.xxl - MIN_SPACE.xxl) / (MAX_WIDTH - MIN_WIDTH),
  xxxl: (MAX_SPACE.xxxl - MIN_SPACE.xxxl) / (MAX_WIDTH - MIN_WIDTH),
  xxxxl: (MAX_SPACE.xxxxl - MIN_SPACE.xxxxl) / (MAX_WIDTH - MIN_WIDTH),
}
// rounded to the nearest 0.25px
const INTERCEPT_SPACE = {
  xxxs: Math.round(4 * (MIN_SPACE.xxxs - SLOPE_SPACE.xxxs * MIN_WIDTH)) / 4,
  xxs: Math.round(4 * (MIN_SPACE.xxs - SLOPE_SPACE.xxs * MIN_WIDTH)) / 4,
  xs: Math.round(4 * (MIN_SPACE.xs - SLOPE_SPACE.xs * MIN_WIDTH)) / 4,
  sm: Math.round(4 * (MIN_SPACE.sm - SLOPE_SPACE.sm * MIN_WIDTH)) / 4,
  md: Math.round(4 * (MIN_SPACE.md - SLOPE_SPACE.md * MIN_WIDTH)) / 4,
  lg: Math.round(4 * (MIN_SPACE.lg - SLOPE_SPACE.lg * MIN_WIDTH)) / 4,
  xl: Math.round(4 * (MIN_SPACE.xl - SLOPE_SPACE.xl * MIN_WIDTH)) / 4,
  xxl: Math.round(4 * (MIN_SPACE.xxl - SLOPE_SPACE.xxl * MIN_WIDTH)) / 4,
  xxxl: Math.round(4 * (MIN_SPACE.xxxl - SLOPE_SPACE.xxxl * MIN_WIDTH)) / 4,
  xxxxl: Math.round(4 * (MIN_SPACE.xxxxl - SLOPE_SPACE.xxxxl * MIN_WIDTH)) / 4,
}
export const spacing = stylex.defineVars({
  xxxs: `clamp(${MIN_SPACE.xxxs}px, calc(${INTERCEPT_SPACE.xxxs}px - ${
    Math.round(10000 * SLOPE_SPACE.xxxs) / 100
  }vw), ${MAX_SPACE.xxxs}px)`,
  xxs: `clamp(${MIN_SPACE.xxs}px, calc(${INTERCEPT_SPACE.xxs}px - ${
    Math.round(10000 * SLOPE_SPACE.xxs) / 100
  }vw), ${MAX_SPACE.xxs}px)`,
  xs: `clamp(${MIN_SPACE.xs}px, calc(${INTERCEPT_SPACE.xs}px - ${
    Math.round(10000 * SLOPE_SPACE.xs) / 100
  }vw), ${MAX_SPACE.xs}px)`,
  sm: `clamp(${MIN_SPACE.sm}px, calc(${INTERCEPT_SPACE.sm}px - ${
    Math.round(10000 * SLOPE_SPACE.sm) / 100
  }vw), ${MAX_SPACE.sm}px)`,
  md: `clamp(${MIN_SPACE.md}px, calc(${INTERCEPT_SPACE.md}px - ${
    Math.round(10000 * SLOPE_SPACE.md) / 100
  }vw), ${MAX_SPACE.md}px)`,
  lg: `clamp(${MIN_SPACE.lg}px, calc(${INTERCEPT_SPACE.lg}px - ${
    Math.round(10000 * SLOPE_SPACE.lg) / 100
  }vw), ${MAX_SPACE.lg}px)`,
  xl: `clamp(${MIN_SPACE.xl}px, calc(${INTERCEPT_SPACE.xl}px - ${
    Math.round(10000 * SLOPE_SPACE.xl) / 100
  }vw), ${MAX_SPACE.xl}px)`,
  xxl: `clamp(${MIN_SPACE.xxl}px, calc(${INTERCEPT_SPACE.xxl}px - ${
    Math.round(10000 * SLOPE_SPACE.xxl) / 100
  }vw), ${MAX_SPACE.xxl}px)`,
  xxxl: `clamp(${MIN_SPACE.xxxl}px, calc(${INTERCEPT_SPACE.xxxl}px - ${
    Math.round(10000 * SLOPE_SPACE.xxxl) / 100
  }vw), ${MAX_SPACE.xxxl}px)`,
  xxxxl: `clamp(${MIN_SPACE.xxxxl}px, calc(${INTERCEPT_SPACE.xxxxl}px - ${
    Math.round(10000 * SLOPE_SPACE.xxxxl) / 100
  }vw), ${MAX_SPACE.xxxxl}px)`,
})
```

```js
import { createTheme } from "@stylexjs/stylex"
import { colors } from "../tokens.stylex"

const DARK = "@media (prefers-color-scheme: dark)"
const LIGHT_TEXT = "#ffffff"
const DARK_TEXT = "#121212"

const lightColors = {
  primaryText: DARK_TEXT,
  primaryTextInverted: LIGHT_TEXT,
  secondaryText: "#c6c6c6",
  secondaryTextInverted: "#c6c6c6",
  lightText: LIGHT_TEXT,
  darkText: DARK_TEXT,
  surface1: "#ffffff",
  surface2: "#c6c6c6",
  surface3: "#5e5e5e",
  surface4: "#303030",
  backgroundImage: "linear-gradient(to bottom, rgb(214, 219, 220), white)",
  border: "#121212",
  primary: "#04b1fb",
  primary1: "#1d8bc3",
  primary2: "#21668f",
  primary3: "#1e445d",
  primary4: "#152530",
  secondary: "#3f7f85",
  secondary1: "#346469",
  secondary2: "#2a4b4e",
  secondary3: "#1f3335",
  secondary4: "#141d1e",
  info: "#388cfa",
  warning: "#cc8800",
  success: "#009f42",
  error: "#b41c2b",
}

const darkColors = {
  primaryText: LIGHT_TEXT,
  primaryTextInverted: DARK_TEXT,
  secondaryText: "#c6c6c6",
  secondaryTextInverted: "#c6c6c6",
  lightText: LIGHT_TEXT,
  darkText: DARK_TEXT,
  surface1: "#121212",
  surface2: "#282828",
  surface3: "#3f3f3f",
  surface4: "#575757",
  backgroundImage: "linear-gradient(to bottom, rgb(20, 22, 27), black)",
  border: "#ffffff",
  primary: lightColors.primary,
  primary1: lightColors.primary1,
  primary2: lightColors.primary2,
  primary3: lightColors.primary3,
  primary4: lightColors.primary4,
  secondary: lightColors.secondary,
  secondary1: lightColors.secondary1,
  secondary2: lightColors.secondary2,
  secondary3: lightColors.secondary3,
  secondary4: lightColors.secondary4,
  info: lightColors.info,
  warning: lightColors.warning,
  success: lightColors.success,
  error: lightColors.error,
}

const autoColors = {
  primaryText: {
    default: lightColors.primaryText,
    [DARK]: darkColors.primaryText,
  },
  primaryTextInverted: {
    default: lightColors.primaryTextInverted,
    [DARK]: darkColors.primaryTextInverted,
  },
  secondaryText: {
    default: lightColors.secondaryText,
    [DARK]: darkColors.secondaryText,
  },
  secondaryTextInverted: {
    default: lightColors.secondaryTextInverted,
    [DARK]: darkColors.secondaryTextInverted,
  },
  lightText: lightColors.lightText,
  darkText: lightColors.darkText,
  surface1: {
    default: lightColors.surface1,
    [DARK]: darkColors.surface1,
  },
  surface2: {
    default: lightColors.surface2,
    [DARK]: darkColors.surface2,
  },
  surface3: {
    default: lightColors.surface3,
    [DARK]: darkColors.surface3,
  },
  surface4: {
    default: lightColors.surface4,
    [DARK]: darkColors.surface4,
  },
  backgroundImage: {
    default: lightColors.backgroundImage,
    [DARK]: darkColors.backgroundImage,
  },
  border: {
    default: lightColors.border,
    [DARK]: darkColors.border,
  },
  primary: lightColors.primary,
  primary1: lightColors.primary1,
  primary2: lightColors.primary2,
  primary3: lightColors.primary3,
  primary4: lightColors.primary4,
  secondary: lightColors.secondary,
  secondary1: lightColors.secondary1,
  secondary2: lightColors.secondary2,
  secondary3: lightColors.secondary3,
  secondary4: lightColors.secondary4,
  info: lightColors.info,
  warning: lightColors.warning,
  success: lightColors.success,
  error: lightColors.error,
}

export const dark = createTheme(colors, darkColors)
export const light = createTheme(colors, lightColors)
export const auto = createTheme(colors, autoColors)
```

**Comment by nmn** [comment]
Let's start by removing `swcMinify: true,` since SWC doesn't work with Babel.

The other thing I need to know is the version of Next you're using.

The issue is within Webpack somewhere. This will likely take a few weeks to resolve.

**Comment by conarwelsh** [comment]
`"next": "^14.0.5-canary.33",` but i can easily switch to a stable version if that helps debug

I removed the swcMinify.  I have still been receiving the error.

No worries on the timeframe, I will update this thread if I find any more relevant info.

**Comment by nmn** [comment]
@conarwelsh Can you try 14.0.1-14.0.3 for now?

I have seen some issues starting with `14.0.4` myself, but I'm unsure what they are.

**Comment by conarwelsh** [comment]
I switched to 14.0.4 but the same problems are occurring.  Some more info for what its worth... it seems most times I save a standard JS file I get the watcher error, but a second save of the same file (or any other file) will clear it.  However if I make a change and save my tokens file, I often need to delete my `.next` folder to clear the error.

Off topic... do you happen to know if the next team plans on porting this setup into their SWC compiler?  Apparently the server actions are not fully supported through the babel compiler at this time, so there are some weird side affects that occur from not using their compiler.

**Comment by nmn** [comment]
The problem with an SWC port is that it will always lag behind the JS version of StyleX. 

At Meta we don't use SWC internally, so we don't have a lot of incentive to maintain a second codebase that we won't use internally and will slow down our iteration speed.

That said, the Next team will port to SWC if there is enough demand.

**Comment by nmn** [comment]
Closing this in favour of #297

**Comment by olivierpascal** [comment]
Same here and I have to say it's pretty annoying, each save requiring a second save (to get rid of the error), and a hard reload (to have new styles applied).

> TypeError: Cannot read properties of undefined (reading 'watcher')

**Comment by olivierpascal** [comment]
@nmn  Would it be possible to know if any effort is currently put into the nextjs plugin to smooth the development flow? I don't know if everyone is experiencing the same issue, or if there is any workaround for this in the meantime, but It's hardly usable now. Thank you very much 🙏🏼

**Comment by nmn** [comment]
Please see the issue #297. It's what I'm tackling next. Once that is done we will be able to go to the next step which is:

A `pre-app` or `pre-src` folder with your actual source code. A `babel-watch` command will transform this folder into `app` or `src` respectively. Which runs in parallel with a standard `swc`-based `next dev` command combined with a single PostCSS or LightningCSS plugin.

Initially this won't support importing styles from `node_modules`, but we will be able to fix that with some import re-writing and configuration as well.
